### DISCUSSION TITLE
Ideas & Suggestions

### DISCUSSION THREAD

**Discussion by sol3uk** [prabhuignoto/react-chrono#286] [Ideas]
I'm working on a small project using this package and it's going great so far, thanks for making it!
I did have a few things that I think could be improved upon, and I'm happy to contribute where I'm able, just wanted to see if these were things that had either been considered before or are maybe already in the pipeline.

# Ideas:
- Styled components support / more in-depth theming
- Ability to re-arrange the order of the components inside Chrono, i.e. I want to be able to move the navigation controls to the bottom of the page when in horizontal view, I've found I can only move the cards.
- Improvements to the outline navigation popup (with large datasets as you scroll you lose the ability to close the popup, maybe onBlur event could be a good fix for this?)
- Horizontal scrolling/dragging support, the feel between mobile (vertical) and desktop (horizontal) can be a bit jarring from being able to scroll to then having to tap to navigate when moving horizontally when a user might expect to be able to swipe to scroll instead.
- Support for item grouping (if the date is the same) we could group events of the same time into a single card. This would save the user from needing to construct custom HTML and building the cards manually.
- Navigation ability to jump forward/back by a set amount like 5 or so, instead of just 1 item at a time or right to the end/beginning of the timeline.

# Issues:
- Not sure if I am using it wrong or something but cardWidth doesn't seem to bring the selected card area down at all, just the interior size of the card, causing these margins and the selection box to float around?
![image](https://user-images.githubusercontent.com/26093813/186277313-46f0c40c-be01-49e0-8e57-f1b9d8842bc9.png)
- The outline seems to be being applied to the wrapper rather than the section that has the border radius on it, this results in hard sharp outlines rather than outlines that follow the curve of the element, not sure if this is a style choice but would be nice to have the option.
Currently where it's applied:
![image](https://user-images.githubusercontent.com/26093813/186280383-46986001-badd-4b2c-a929-154481a99148.png)   ![image](https://user-images.githubusercontent.com/26093813/186280405-bba166f0-e850-40c3-95a0-dcad2b7a17d8.png)
Alternative option: 
![image](https://user-images.githubusercontent.com/26093813/186280330-9ff3d60a-3d64-4895-9671-3934ffc9a9da.png)   ![image](https://user-images.githubusercontent.com/26093813/186280272-11aad0f6-5170-4f94-92bb-c7731644a202.png)

Or also the same border radius can be applied to the wrapper.

I'll probably have more over the next few days, again, thanks for developing this and for reading my ideas/feedback!

**Comment by prabhuignoto** [comment]
@sol3uk 

Thanks for taking the time to write such a detailed feedback. Thank you very much.

I'm sure that the ideas will add a lot to the library. I will tell you the plan for how to move forward with this and put the steps in place in stages.

At the moment, I am the only person working on the project. Please don't hesitate to let me know if you want to add a certain feature yourself, and we can plan it accordingly

**Comment by strikingly-william** [comment]
@prabhuignoto Can we support image preview function?

**Reply by prabhuignoto** [reply]
@strikingly-william If I understand correctly, you would like the images within the timeline cards to be enlarged and displayed in a more prominent manner, is that correct?
